modules/jaas/src/test/java/org/mule/module/jaas/JaasAutenticationWithJaasConfigFileTestCase.java

```diff
@@ -29,59 +30,46 @@ protected String getConfigFile()
     @Test
     public void goodAuthentication() throws Exception
     {
-        Map<String, Object> props = createMessagePropertiesWithCredentials("Marie.Rizzo", "dragon");
-        MuleMessage m = muleContext.getClient().send("vm://test", "Test", props);
+        SecurityHeader securityHeader = createSecurityHeader("Marie.Rizzo", "dragon");
+        MuleEvent result = flowRunner("TestUMO").withInboundProperty(securityHeader.getKey(), securityHeader.getValue()).withPayload("Test").run();
 
-        assertNotNull(m);
-        assertTrue(m.getPayload() instanceof String);
-        assertEquals("Test Received", getPayloadAsString(m));
+        assertNotNull(result);
+        assertTrue(result.getMessage().getPayload() instanceof String);
+        assertEquals("Test Received", getPayloadAsString(result.getMessage()));
     }
 
     @Test
     public void anotherGoodAuthentication() throws Exception
     {
-        Map<String, Object> props = createMessagePropertiesWithCredentials("anon", "anon");
-        MuleMessage m = muleContext.getClient().send("vm://test", "Test", props);
+        SecurityHeader securityHeader = createSecurityHeader("anon", "anon");
+        MuleEvent result = flowRunner("TestUMO").withInboundProperty(securityHeader.getKey(), securityHeader.getValue()).withPayload("Test").run();
 
-        assertNotNull(m);
-        assertTrue(m.getPayload() instanceof String);
-        assertEquals("Test Received", getPayloadAsString(m));
+        assertNotNull(result);
+        assertTrue(result.getMessage().getPayload() instanceof String);
+        assertEquals("Test Received", getPayloadAsString(result.getMessage()));
     }
 
     @Test
     public void wrongCombinationOfCorrectUsernameAndPassword() throws Exception
     {
-        Map<String, Object> props = createMessagePropertiesWithCredentials("Marie.Rizzo", "anon");
-
-        MuleMessage message = muleContext.getClient().send("vm://test", "Test", props);
-        assertNotNull(message);
-        assertNotNull(message.getExceptionPayload());
-        assertTrue(ExceptionUtils.containsType(message.getExceptionPayload().getException(),
-            UnauthorisedException.class));
+        SecurityHeader securityHeader = createSecurityHeader("Marie.Rizzo", "anon");
+        expectedException.expect(UnauthorisedException.class);
+        flowRunner("TestUMO").withInboundProperty(securityHeader.getKey(), securityHeader.getValue()).withPayload("Test").run();
```

Comment:
You can use runExpectingException(), which will return the exceptions to do the assertions on.